src/main/java/com/zufar/onlinestore/user/converter/UserDtoConverter.java

```diff
@@ -1,65 +1,54 @@
 package com.zufar.onlinestore.user.converter;
 
-import com.zufar.onlinestore.user.dto.AddressDto;
 import com.zufar.onlinestore.user.dto.UserDto;
-import com.zufar.onlinestore.user.entity.Address;
 import com.zufar.onlinestore.user.entity.Authority;
 import com.zufar.onlinestore.user.entity.UserEntity;
 import com.zufar.onlinestore.user.entity.UserGrantedAuthority;
-import lombok.AllArgsConstructor;
-import org.springframework.stereotype.Service;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.factory.Mappers;
 
 import java.util.Collections;
 import java.util.Set;
 
-@Service
-@AllArgsConstructor
-public class UserDtoConverter {
-    private final AddressDtoConverter addressDtoConverter;
+@Mapper(componentModel = "spring", uses = AddressDtoConverter.class)
+public interface UserDtoConverter {
 
-    public UserDto toDto(final UserEntity entity) {
-        AddressDto addressDto = null;
-        if (entity.getAddress() != null) {
-            addressDto = addressDtoConverter.toDto(entity.getAddress());
-        }
-        return new UserDto(
-                entity.getUserId(),
-                entity.getFirstName(),
-                entity.getLastName(),
-                entity.getUsername(),
-                entity.getEmail(),
-                entity.getPassword(),
-                addressDto
-        );
-    }
+    AddressDtoConverter addressConverter = Mappers.getMapper(AddressDtoConverter.class);
```

Comment:
it seems like a bad practice to add public static final field here

Reply:
I guess you have to add abs class instead of doing this here

Reply:
fixed